### PR TITLE
Add a zeroing step to filters

### PR DESCRIPTION
**Pull request by SylviaWhittle**
This PR is just to add a height zeroing step before masking, as previously using an absolute threshold on **some** images was being affected by the image being too high up / low down and not centred on its average height. This is mostly a "just in case" step that doesn't add any time to the overall computation time of `filters.py`.

All it does is adds a new `self.average_background` step before masking in `filters.py`, and renames the plot names to `initial_zero_average_background` and `final_zero_average_background`.

Also removed some debugging lines that I forgot to remove in the last PR.

### CODE REVIEW

**Comment by SylviaWhittle** [comment]
CodeCov doesn't seem to like me removing the logging lines... Apparently removing lines reduces coverage?

**Review by ns-rse — CHANGES_REQUESTED** [review]
Baffled why the coverage changes removing logging statements.

Not sure if we should be switching the `image_set: all` in `topostats/default_config.yaml` as it introduces lots of additional processing time.

File: topostats/default_config.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ plotting:
   run: true # Options : true, false
   save_format: png # Options : see https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html
   pixel_interpolation: null # Options : https://matplotlib.org/stable/gallery/images_contours_and_fields/interpolation_methods.html
-  image_set: core  # Options : all, core
+  image_set: all  # Options : all, core
```
[the remark this answers is not in the document]
Comment (SylviaWhittle) [reply topostats/default_config.yaml]: This was accidental. I missed it when sanity checking my PR, sorry! Reverted

File: topostats/validation.py
[the remark this answers is not in the document]
Comment (SylviaWhittle) [reply topostats/validation.py:1]: Yeah I'll look at my local / pre-commit black settings to see if there's a difference.

File: topostats/plotting.py [hunk 2]
```diff
@@ -157,9 +157,6 @@ def sns_plot(self) -> Optional[Tuple[plt.Figure, plt.Axes]]:
         # The try/catch is there to catch this error and skip plotting KDEs if all values are the same.
 
         fig, ax = self._setup_figure()
-        LOGGER.info(f"melted data: \n {self.melted_data}")
```
[the remark this answers is not in the document]
Comment (SylviaWhittle) [reply topostats/plotting.py:160]: Do you know how to get around this issue? I don't necessarily think these lines are particularly useful.

Here is a post talking about it but with no resolutions yet: https://community.codecov.com/t/project-coverage-decrease-on-deleted-line/2726/3

It technically is a decrease in coverage as removing covered lines will decrease the proportion. Perhaps not allowing PRs that decrease coverage is not useful and instead we ensure that all added lines are covered?

[the remark this answers is not in the document]
Comment (ns-rse) [reply topostats/plotting.py:160]: I'm not convinced its an issue with removing lines, rather that they should not have been included in the coverage anyway as unless we have a specific test that checks for the output in the logs there is no coverage of those lines. Its a false positive so happy for this to be merged.

**Review by ns-rse — APPROVED** [review]
Thanks for fixing up `default_config.yaml` :+1:

**Pull request merged**